docs/fud/synthesis.md

```diff
@@ -1,44 +1,152 @@
+
+    fud e --to hls-estimate examples/dahlia/dot-product.fuse
+
+There is also an `hls-files` state for the raw results of Vivado HLS.
+
+## Emulation and Execution
+
+`fud` can also compile Calyx programs for actual execution, either in the Xilinx toolchain's emulation modes or for running on a physical FPGA.
+This route involves generating an [AXI][] interface wrapper for the Calyx program and invoking it using [XRT][]'s OpenCL interface.
+
+### Set Up
+
+As above, you can invoke the Xilinx toolchain locally or remotely, via SSH.
+To set up SSH execution, you can edit your `config.toml` to add settings like this:
+
+    [stages.xclbin]
+    ssh_host = "havarti"
+    ssh_username = "als485"
+    remote = true
+
+To use local execution, just leave off the `remote = true` line.
+
+You can also set the Xilinx mode and target device:
+
+    [stages.xclbin]
+    mode = "hw_emu"
+    device = "xilinx_u50_gen3x16_xdma_201920_3"
+
+The options for `mode` are `hw_emu` (simulation) and `hw` (on-FPGA execution).
+The device string above is for the [Alveo U50][u50] card, which we have at Cornell, but I honestly don't know how you're supposed to find the right string for a different FPGA target.
+Hopefully someone will figure this out and document it in the future.
+
+To use hardware emulation, you will also need to configure the `wdb` stage.
+It has similar `ssh_host`, `ssh_username`, and `remote` options to the `xclbin` stage.
+You will also need to configure the stage to point to your installations of [Vitis][] and [XRT][], like this:
+
+    [stages.wdb]
+    xilinx_location: /scratch/opt/Xilinx/Vitis/2020.2
+    xrt_location: /opt/xilinx/xrt
+
+### Compile
+
+The first step in the Xilinx toolchain is to generate [an `xclbin` executable file][xclbin].
+Here's an example of going all the way from a Calyx program to that:
+
+    fud e examples/futil/dot-product.futil -o foo.xclbin
+
+On our machines, compiling even a simple example like the above for simulation takes about 5 minutes, end to end.
+A failed run takes about 2 minutes to produce an error.
+
+By default, the Xilinx tools run in a temporary directory that is deleted when `fud` finishes.
+To instead keep the sandbox directory, use `-s xclbin.save_temps true`.
+You can then find the results in a directory named `fud-out-N` for some number `N`.
+
+### Emulate
+
+You can also execute compiled designs through Xilinx hardware emulation.
+Use the `wdb` state as your `fud` target:
+
+    fud e -vv foo.xclbin -s wdb.save_temps true -o out.wdb
+
+This stage produces a Vivado [waveform database (WDB) file][wdb]
+Through the magic of `fud`, you can also go all the way from a Calyx program to a `wdb` file in the same way.
+There is also a `wdb.save_temps` option, as with the `xclbin` stage.
+
+You also need to provide a host C++ program via the `wdb.host` parameter, but I don't know much about that yet, so documentation about that will have to wait.
```

Comment:
It might be useful to provide a TODO with an Issue in case a reader passing by is familiar with this :) (similar for the part about the FPGA string above).

Reply:
A good point… however, I actually think I might end up advocating for dropping this fud stage altogether in favor of a different one (which will be the subject of a future PR), so I might end up deleting this bit in the docs anyway.